ci-bench/src/bench_lib/compare.rs

```diff
@@ -0,0 +1,188 @@
+use anyhow::Context;
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::path::Path;
+
+/// The threshold at which instruction count changes are considered relevant
+const CHANGE_THRESHOLD: f64 = 0.002; // 0.2%
+
+/// The results of a comparison between two `run-all` executions
+struct CompareResult {
+    diffs: Vec<Diff>,
+    /// Benchmark scenarios present in the candidate but missing in the baseline
+    missing_in_baseline: Vec<String>,
+}
+
+/// Contains information about instruction counts and their difference for a specific scenario
+struct Diff {
+    scenario: String,
+    baseline: u64,
+    candidate: u64,
+    diff: i64,
+    diff_ratio: f64,
+}
+
+/// Compares all benchmark results between a baseline and a candidate, printing an overview to
+/// stdout
```

Comment:
tiny nit: missing trailing `.` :-)
```suggestion
/// stdout.
```